sunpy/map/header_helper.py

```diff
@@ -361,3 +362,54 @@ def _set_instrument_meta(meta_wcs, instrument, telescope, observatory, wavelengt
     'CD2_2':
     'Matrix element CDi_j describing the rotation required to align solar North with the top of the image.'
 }
+
+
+def carrington_header(date, observer_coordinate, shape, *, projection_code="CAR"):
```

Comment:
I would like to bikeshed this a little.

What would be the pros/cons of making this into a generic heliographic helper, where it can do stonyhurst as well?

It would make the API slightly more complex, i.e adding a new keyword argument for frame, but on the pro side it would allow people to reuse the pixel size calculation for stonyhurst frames as well?

Reply:
What's your proposed API, including a new function name? I'm not against making it more general in theory, but think there is value in having a simple and recognisable function name.

Perhaps this is where some sort of user testing/questioning would be valuable?

Reply:
I was thinking something like:
```python
def make_heliographic_header(date, observer_coordinate, shape, *, frame='heliographic_carrington', projection_code="CAR")
```
where frame is then passed to SkyCoord to make the reference coordinate?

Reply:
Hmm, I would say that "Carrington" is a more recognisable name than "Heliographic" to a general solar physicist, so I'm pro keeping it as Carrington. If there's demand for a Stonyhurst version then we could add `make_stonyhurst_header` and de-duplicate the interal logic instead of unifying the API?

Reply:
Decided to go with `heliographic` in the community meeting today